[IMP] server: intercept SIGXCPU and shutdown gracefully

d-fence opened the pull request:
When the runbot spawns an Odoo instance, a CPU TIME limit is set.
When this limit is reached, the kernel kills the Odoo instance and no
message appears in the logs leaving the dev in a deep smoke cloud.

With this commit, the signal sent by the kernel when the soft
limit is reached (SIGXCPU), is intercepeted and Odoo shutdown gracefully with a
nice message for the dev.

Description of the issue/feature this PR addresses: When using the code coverage, the test build time is increased. So, it happens that the soft limit on CPU time, set by the runbot, is reached and the build is killed without any message.

Current behavior before PR: When Odoo is killed by the kernel when reaching the soft limit CPU time, the shutdown is brutal.

Desired behavior after PR is merged: When Odoo is killed by the kernel when reaching the soft limit on CPU time, Odoo shutdown gracefully.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

odony commented:
Possibly you could retarget to saas-11.2/3 if you want to cover more branches on runbot ;-)

d-fence commented:
Ok, changed as requested and reset hard to saas-11.2.
Also changed the base target of the github PR to saas-11.2.
The only problem that I could not solve is that the branch name starts with "master-"

mreficent commented on odoo/service/server.py:
```diff
@@ -207,6 +207,9 @@ def signal_handler(self, sig, frame):
             # interrupt run() to start shutdown
             raise KeyboardInterrupt()
+        elif sig == signal.SIGXCPU:
+            sys.stderr.write("CPU time limit exceeded ! Shutting down immediately\n")
```
Why the space between `exceeded` and `!` symbol?
